provider.go

```diff
@@ -301,6 +294,13 @@ func (p *Provider) MountSecretsStoreObjectContent(ctx context.Context, attrib ma
 	if tenantID == "" {
 		return fmt.Errorf("tenantId is not set")
 	}
+
+	azureCloudEnv, err := ParseAzureEnvironment(cloudName)
```

Comment:
Can we add a unit test for `ParseAzureEnvironment` to check we are getting the azure envs as expected for different cloud names?

Reply:
Added unit tests